### PR TITLE
(#293) Solved factor config_open_default_target out of wrapper.h

### PR DESCRIPTION
**Pull request by rmknan**
The issue has been solved and the test run successfully. Made changes to` stumpless.yml `and deleted `private/config/wrapper.h`. New header file `open_default_target.h` created.

### CODE REVIEW

**Comment by rmknan** [comment]
#293

**Comment by rmknan** [comment]
I am on a different branch but it seems yesterday's commit have been re-committed. Is that fine?

File: include/private/config/wrapper/open_default_target.h [hunk 1]
```diff
@@ -1,26 +1,27 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 
-#ifndef __STUMPLESS_PRIVATE_CONFIG_WRAPPER_H
-#  define __STUMPLESS_PRIVATE_CONFIG_WRAPPER_H
+#ifndef __STUMPLESS_PRIVATE_CONFIG_WRAPPER_CONFIG_OPEN_DEFAULT_TARGET_H
```
Comment (goatshriek) [inline include/private/config/wrapper/open_default_target.h]: update the header guard to match the filename (remove the extra `CONFIG`)

File: include/private/config/wrapper/open_default_target.h [hunk 2]
```diff
@@ -38,5 +39,4 @@
 #    define config_open_default_target file_open_default_target
 #    define config_close_default_target stumpless_close_file_target
 #  endif
-
-#endif /* __STUMPLESS_PRIVATE_CONFIG_WRAPPER_H */
+#endif /* __STUMPLESS_PRIVATE_CONFIG_OPEN_DEFAULT_TARGET_H */
```
Comment (goatshriek) [inline include/private/config/wrapper/open_default_target.h]: make sure that the comment matches the header guard name

File: src/target.c [hunk 3]
```diff
@@ -39,7 +39,8 @@
 #include "private/config.h"
 #include "private/config/network_support_wrapper.h"
 #include "private/config/locale/wrapper.h"
-#include "private/config/wrapper.h"
+// #include "private/config/wrapper.h"
```
Comment (goatshriek) [inline src/target.c]: remove this commented out code

**Review by goatshriek — CHANGES_REQUESTED** [review]
Thank you for putting this together! There are a few minor changes to make, listed below.

Also, be sure to merge in the `latest` branch to yours to remove the seemingly duplicate changes from the string formatting change.

**Comment by rmknan** [comment]
> Also, be sure to merge in the `latest` branch to yours to remove the seemingly duplicate changes from the string formatting change.

Can you elaborate? For this I issue I created a new branch and after the changes I did `git push -u origin <branch>`. Are you asking me to do  g`it push -u origin latest`? or somthing else?

File: include/private/config/wrapper/open_default_target.h (hunk 1 above)
Comment (rmknan) [reply include/private/config/wrapper/open_default_target.h]: So, the name of the header has to be

`#__STUMPLESS_PRIVATE_CONFIG_WRAPPER_OPEN_DEFAULT_TARGET_H` ?

Comment (goatshriek) [reply include/private/config/wrapper/open_default_target.h]: The header guard symbol should be named `__STUMPLESS_PRIVATE_CONFIG_WRAPPER_OPEN_DEFAULT_TARGET_H`, yes.

**Comment by goatshriek** [comment]
You need to perform a pull before you do another push. Exactly how you do the pull depends on how you have your local repository set up, but it's likely something like: `git pull upstream latest`. If you don't have a remote set up named upstream for this repository, you can do so with this command: `git remote add upstream git@github.com:goatshriek/stumpless.git`. If you run `git log` and see the commit regarding param to string format, you'll know that you have succeeded.

**Pull request merged**